main.cpp

```diff
@@ -56,14 +56,14 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const std::string c) : value(v), name(c) {}   //1
+    int value;
+    std::string name; 
 };
 
-struct <#structName1#>                                //4
+struct struct1                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
         if( a->value < b->value ) return a;
         if( a->value > b->value ) return b;
```

Comment:
What do we always check before using pointers? 
- [ ] revise

Reply:
Always check to make sure your pointer is not NULL before using it!!!! :)